Truncate in template_comparison by default rather than nan_fill

rosteen opened the pull request:
Implements the suggestion given in #653. One thing I changed here, which I'm not sure of, is that the returned template is now resampled onto the observed spectrum's spectral axis, in addition to being normalized as previous. Opening as draft since I still need to update the docs, and I'll probably need to add a bit more test coverage.

camipacifici reviewed (commented):
Tested now. I see it uses only the region of overlap between the template and the observed spectrum to calculate the chi^2, but it also returns the template only in the overlapping region, while it would be useful to get out the full original template properly redshifted and normalized. Here are two screenshots: one shows the observed spectrum and one of the template in its full range; the other shows the observed spectrum and the selected template after template_comparison.
![Screenshot 2024-02-06 at 12 03 46 PM](https://github.com/astropy/specutils/assets/19998720/87c6bacd-eddb-49c9-9c8d-b40e8d7bee30)
![Screenshot 2024-02-06 at 12 03 56 PM](https://github.com/astropy/specutils/assets/19998720/a7a638e2-fb25-4aeb-9f8e-7ab5a232d6e1)

rosteen commented:
@camipacifici Thanks for the comment, I reverted back to returning the full template. I also fixed an existing bug in the redshifting so that it will now work on any unit of spectral axis (e.g., frequency), not just wavelengths.

camipacifici reviewed (approved):
Functionality looks good to me!
